modules/update-evaluate-availability.adoc

```diff
@@ -0,0 +1,113 @@
+// Module included in the following assemblies:
+//
+// * updating/understanding_updates/how-updates-work.adoc
+
+:_content-type: CONCEPT
+[id="update-evaluate-availability_{context}"]
+= Evaluating update availability
+
+The Cluster Version Operator (CVO) periodically queries the OpenShift Update Service for the most recent graph of update possibilities.
+This graph is based on the channel to which the cluster is subscribed.
+If the graph contains paths to OCP versions that are newer than the cluster's current version, the CVO saves these versions into either the `availableUpdates` or `conditionalUpdates` field of its ClusterVersion resource.
```

Comment:
nit: there are a few "OCP version(s)" references.  Should they be `{product-title} versions`?  Or `OpenShift Container Platform (OCP) version` the first time and `OCP version` afterwards?  Or something?  I expect most readers will be able to unpack OCP, but it still makes me a bit jittery to open with a collapsed acronym ;).

Reply:
"newer than the cluster's current version" isn't quite right.  For example:

```console
openshift/cincinnati-graph-data $ hack/show-edges.py --cincinnati https://api.openshift.com/api/upgrades_info/graph fast-4.13 | grep '^4[.]12[.]0 -\|4[.]13[.]1'
4.12.0 -(blocked: ConsoleAvailableUpdatesNull)-> 4.12.1
4.12.0 -(blocked: ConsoleAvailableUpdatesNull)-> 4.12.1
4.12.0 -> 4.12.10
4.12.0 -> 4.12.11
4.12.0 -(blocked: PatchesOlderRelease)-> 4.12.12
4.12.0 -> 4.12.13
4.12.0 -> 4.12.14
4.12.0 -> 4.12.15
4.12.0 -> 4.12.16
4.12.0 -> 4.12.17
4.12.0 -> 4.12.18
4.12.0 -> 4.12.19
4.12.0 -(blocked: LeakedMachineConfigBlocksMCO)-> 4.12.2
4.12.0 -(blocked: LeakedMachineConfigBlocksMCO)-> 4.12.3
4.12.0 -(blocked: LeakedMachineConfigBlocksMCO)-> 4.12.4
4.12.0 -(blocked: LeakedMachineConfigBlocksMCO)-> 4.12.5
4.12.0 -(blocked: LeakedMachineConfigBlocksMCO)-> 4.12.6
4.12.0 -(blocked: LeakedMachineConfigBlocksMCO)-> 4.12.7
4.12.0 -> 4.12.8
4.12.0 -> 4.12.9
4.12.16 -> 4.13.1
4.12.17 -> 4.13.1
4.12.18 -> 4.13.1
4.12.19 -> 4.13.1
4.13.0 -> 4.13.1
```

Folks currently on 4.12.0 and in channel `fast-4.13` have 4.13.1 as a "newer than" release (both in terms of build date and [SemVer ordering][1].  But because of [some 4.12.z changes we want folks to pick up before they head to 4.13][2], there are no direct update recommendations from 4.12.0 straight to 4.13.1.  Instead, we recommend chains like 4.12.0 -> 4.12.19 -> 4.13.1.

If we instead [pivot to talking about update recommendations][3], we can skip right to populating `availableUpdates` and `conditionalUpdates`, without having to invent wording to bridge between "graph paths" and those update recommendations.

[1]: https://semver.org/spec/v2.0.0.html#spec-item-11
[2]: https://github.com/openshift/cincinnati-graph-data/blame/00c9be09fff5339fc26493dd88a921c2df282334/build-suggestions/4.13.yaml#L2
[3]: https://github.com/openshift/openshift-docs/pull/60091#discussion_r1212343513

Reply:
That's a good catch, the first mention of OCP on this page should be written out.

Reply:
Here's my suggested rewrite to resolve both this comment about "newer than" releases and limiting the mention of "graphs" per your comment below this one:

> The Cluster Version Operator (CVO) periodically queries an OpenShift Update Service for the most recent data about update possibilities. This data is based on the channel to which the cluster is subscribed. The CVO saves update recommendations and risks into either the availableUpdates or conditionalUpdates field of its ClusterVersion resource.